Client/src/ClientSceneManager/ClientSceneManager.cpp

```diff
@@ -0,0 +1,104 @@
+
+SceneManager::ClientSceneManager::ClientSceneManager(std::shared_ptr<std::vector<ECS::Registry>> registries)
+{
+    _registries = registries;
+    _registerIndex = SceneManager::RegisterIndex::CURRENT;
+
+    ECS::Registry reg;
+    _registries->push_back(reg);
+
+    _keysRegistry.push_back(std::unordered_map<KEY_MAP, std::shared_ptr<ISystem>>());
+
+    _loadScene(DEFAULT_SCENE, _registerIndex);
+}
+
+void SceneManager::ClientSceneManager::_loadScene(const std::string &path, std::size_t registerIndex)
+{
+    std::ifstream file(SCENE_PATH + path);
+    Json::Reader reader;
+    Json::Value root;
+
+    if (!reader.parse(file, root, false))
+        throw SceneManagerError("Error while parsing the scene file: " + path);
+    _loadSceneComponents(root, registerIndex);
+    _loadSceneSystems(root, registerIndex);
+    _loadSceneKeys(root, registerIndex);
+}
+
+void SceneManager::ClientSceneManager::_loadSceneComponents(Json::Value root, std::size_t index)
+{
+    const Json::Value& entities = root["entities"];
+
+    for (unsigned int i = 0; i < entities.size(); ++i) {
+        const Json::Value& componentList = entities[i];
+        ECS::entity_t entity = _registries->at(index).spawn_entity();
+        for (unsigned int j = 0; j < componentList.size(); ++j) {
+            std::shared_ptr<IComponent> component = DLLoader<IComponent>::load(LIB_COMPONENTS_PATH + componentList[j].asString(), "loadComponentInstance");
+            if (component) {
+                _registries->at(index).register_component<IComponent>(component->getType());
+                _registries->at(index).set_component(entity, component, component->getType());
+            } else {
+                throw SceneManagerError("Error while loading the component: " + componentList[j].asString());
+            }
+        }
+    }
+}
+
+void SceneManager::ClientSceneManager::_loadSceneSystems(Json::Value root, std::size_t index)
+{
+    const Json::Value& systems = root["systems"];
+
+    for (unsigned int i = 0; i < systems.size(); ++i) {
+        std::shared_ptr<ISystem> system = DLLoader<ISystem>::load(LIB_SYSTEMS_PATH + systems[i].asString(), "loadSystemInstance");
+        if (system)
+            _registries->at(index).add_system(system->getFunction());
+        else
+            throw SceneManagerError("Error while loading the system: " + systems[i].asString());
+    }
+}
+
+void SceneManager::ClientSceneManager::_loadSceneKeys(Json::Value root, std::size_t index)
+{
+    const Json::Value& keyBinds = root["keyBinds"];
+
+    for (auto keys : keyBinds) {
+        const Json::Value &key = keys["key"];
+        const Json::Value &path = keys["path"];
+
+        if (key and path) {
+            if (index == 1 && path.asString().find(CONFIG_SUFFIX) != std::string::npos) {
+                _registerIndex++;
+                ECS::Registry reg;
+                _registries->push_back(reg);
+                _loadScene(path.asString(), _registerIndex);
+            }
```

Comment:
You also need to create a new map for the _KeysRegistries, as the _registries when loading a new scene.

Reply:
Done 👍